gandalf.js

```diff
@@ -1,51 +1,45 @@
 /**
```

Comment:
Consider extracting this repeated logic into a separate function to avoid duplication.